sass-eyeglass/eyeglass-dev-testutils
# dart-sass compatibility

# Summary
eyeglass supports sass (dart-sass). This is an update to eyeglass-dev-testutils to test against dart-sass.

  - [x] Bump `eyeglass@3.0.3`
  - [x] Fix gulp issues.
  - [x] Add dart-sass
- [x] Setup test harness for dart-sass
- [x] Fix fixtures output with trailing newline.
- [x] Needs dropping of third parameter of `name` in render calls.
- [x] Fix compile assertCompiles and compile API methods. 
- [x] Added options.formatter to TestUtils constructor so that node-sass output can be formatted to match that of dart-sass.
  - node-sass will output will not add a newline to the closing brace for style expressions. 
   `dart-sass`
    ```css
    .foo {
      color: #FFF;
    }
    ```
    `node-sass`
    ```css
    .foo {
      color: #FFF;}
    ```
# How to test
Use http://volta.sh 

```
$ yarn install
$ yarn test
```

```
➜  eyeglass-dev-testutils git:(dart-sass-compat) yarn test
yarn run v1.22.4
$ ./node_modules/.bin/gulp test
[13:43:25] Using gulpfile ~/Workspace/os/eyeglass-dev-testutils/gulpfile.js
[13:43:25] Starting 'lint'...
[13:43:25] Finished 'lint' after 126 ms
[13:43:25] Starting 'test'...


  engine: sass
    #getSassFixtures
      ✓ should return the collection of fixtures (keys)
      ✓ should return the collection of fixtures (source)
      ✓ should return the collection of fixtures (expected)
    #assertCompiles
      should compile and assert each fixture
        Compile Fixture `empty`
          ✓ the output should match empty.css
        Compile Fixture `simple`
          ✓ the output should match simple.css

  engine: node-sass
    #getSassFixtures
      ✓ should return the collection of fixtures (keys)
      ✓ should return the collection of fixtures (source)
      ✓ should return the collection of fixtures (expected)
    #assertCompiles
      should compile and assert each fixture
        Compile Fixture `empty`
          ✓ the output should match empty.css
        Compile Fixture `simple`
          ✓ the output should match simple.css


  10 passing (71ms)

[13:43:26] Finished 'test' after 306 ms
✨  Done in 1.20s.
```
